bellatrix.core/src/main/java/solutions/bellatrix/core/plugins/UsesPlugins.java

```diff
@@ -9,6 +9,12 @@ public <T extends Plugin> void addPlugin(Class<T> plugin, Object... args) {
         PluginExecutionEngine.addPlugin(SingletonFactory.getInstance(plugin, args));
     }
 
+    public <T extends Plugin> void addPluginWithKey(Class key, Class<T> plugin, Object... args) {
```

Comment:
what is the idea here - so far the framework only had one method for adding plugins and listeners. What is the need for another one?

Reply:
> what is the idea here - so far the framework only had one method for adding plugins and listeners. What is the need for another one?

`ScreenshotPlugin `is extended by `DesktopScreenshotPlugin`, `MobileScreenshotPlugin`, and `WebScreenshotPlugin`. Inside the base test classes (`AndroidTest`, `DesktopTest`, `IOSTest`, `WebTest`), instead of registering the Screenshot plugins for each respective module by themselves (as their class as key), we register them by the base class - `ScreenshotPlugin`.
This is because VRT plugin doesn't have insight which module calls it; It only expects that the module has an implementation of the `ScreenshotPlugin`. That's why, it searches inside SingletonFactory's map by the key `ScreenshotPlugin`.